gui/petitions.lua

```diff
@@ -0,0 +1,307 @@
+-- Show fort's petitions, pending and fulfilled.
+--[====[
+
+gui/petitions
+=============
+Show fort's petitions, pending and fulfilled.
+
+For best experience add following to your ``dfhack*.init``::
+
+    keybinding add Alt-P@dwarfmode/Default gui/petitions
+
+]====]
+
+local gui = require 'gui'
+local utils = require 'utils'
+
+-- local args = utils.invert({...})
+
+--[[
+[lua]# @ df.agreement_details_type
+<type: agreement_details_type>
+0                        = JoinParty
+1                        = DemonicBinding
+2                        = Residency
+3                        = Citizenship
+4                        = Parley
+5                        = PositionCorruption
+6                        = PlotStealArtifact
+7                        = PromisePosition
+8                        = PlotAssassination
+9                        = PlotAbduct
+10                       = PlotSabotage
+11                       = PlotConviction
+12                       = Location
+13                       = PlotInfiltrationCoup
+14                       = PlotFrameTreason
+15                       = PlotInduceWar
+]]
+
+if not dfhack.world.isFortressMode() then return end
+
+-- from gui/unit-info-viewer.lua
+do -- for code folding
+--------------------------------------------------
+---------------------- Time ----------------------
```

Comment:
Is this section copied from somewhere else? Asking partly due to the different indentation. It might be useful to include in a library so that other scripts can use it too.

Reply:
See comment on line 42: `from gui/unit-info-viewer.lua`